### PR TITLE
alternator/expressions.g: Fix antlr3 missing token leak

### LINKED ISSUE #25878: Alternator: expression parsing memory leak
I stumbled across an issue, where a new SEASTAR_TEST_CASE reports a memory leak:

```cpp
SEASTAR_TEST_CASE(memory_leak) {
    BOOST_REQUIRE_THROW(alternator::parse_projection_expression("a[0.b]"), alternator::expressions_syntax_error);
    co_return;
}
```

```
Direct leak of 88 byte(s) in 1 object(s) allocated from:
    #0 0x0000033bc8b4 in malloc /mnt/clang_build/llvm-project-x86_64/compiler-rt/lib/asan/asan_malloc_linux.cpp:67:3
    #1 0x00000aadb709 in antlr3::DefaultAllocPolicy::alloc(unsigned long) /usr/include/antlr3memory.hpp:137:13
    #2 0x00000aadb709 in antlr3::DefaultAllocPolicy::operator new(unsigned long) /usr/include/antlr3memory.hpp:115:13
    #3 0x00000aadb709 in antlr3::Parser<antlr3::Traits<alternator::expressionsLexer, alternator::expressionsParser, antlr3::CustomTraitsBase, antlr3::Empty>>::getMissingSymbol(antlr3::TokenIntStream<antlr3::Traits<alternator::expressionsLexer, alternator::expressionsParser, antlr3::CustomTraitsBase, antlr3::Empty>>*, antlr3::ANTLR_ExceptionBase<antlr3::Traits<alternator::expressionsLexer, alternator::expressionsParser, antlr3::CustomTraitsBase, antlr3::Empty>, antlr3::CommonTokenStream<antlr3::Traits<alternator::expressionsLexer, alternator::expressionsParser, antlr3::CustomTraitsBase, antlr3::Empty>>>*, unsigned int, antlr3::BitsetList<antlr3::TraitsBase<antlr3::CustomTraitsBase>>*) /usr/include/antlr3parser.inl:214:10
    #4 0x00000aad64ba in antlr3::BaseRecognizer<antlr3::Traits<alternator::expressionsLexer, alternator::expressionsParser, antlr3::CustomTraitsBase, antlr3::Empty>, antlr3::CommonTokenStream<antlr3::Traits<alternator::expressionsLexer, alternator::expressionsParser, antlr3::CustomTraitsBase, antlr3::Empty>>>::getMissingSymbol(antlr3::TokenIntStream<antlr3::Traits<alternator::expressionsLexer, alternator::expressionsParser, antlr3::CustomTraitsBase, antlr3::Empty>>*, antlr3::Traits<alternator::expressionsLexer, alternator::expressionsParser, antlr3::CustomTraitsBase, antlr3::Empty>::ExceptionBaseType<antlr3::CommonTokenStream<antlr3::Traits<alternator::expressionsLexer, alternator::expressionsParser, antlr3::CustomTraitsBase, antlr3::Empty>>>*, unsigned int, antlr3::BitsetList<antlr3::TraitsBase<antlr3::CustomTraitsBase>>*) /usr/include/antlr3baserecognizer.inl:795:28
    #5 0x00000aad64ba in antlr3::BaseRecognizer<antlr3::Traits<alternator::expressionsLexer, alternator::expressionsParser, antlr3::CustomTraitsBase, antlr3::Empty>, antlr3::CommonTokenStream<antlr3::Traits<alternator::expressionsLexer, alternator::expressionsParser, antlr3::CustomTraitsBase, antlr3::Empty>>>::recoverFromMismatchedToken(unsigned int, antlr3::BitsetList<antlr3::TraitsBase<antlr3::CustomTraitsBase>>*) /usr/include/antlr3baserecognizer.inl:482:25
    #6 0x00000aab4cb4 in antlr3::Parser<antlr3::Traits<alternator::expressionsLexer, alternator::expressionsParser, antlr3::CustomTraitsBase, antlr3::Empty>>::matchToken(unsigned int, antlr3::BitsetList<antlr3::TraitsBase<antlr3::CustomTraitsBase>>*) /usr/include/antlr3parser.inl:325:26
    #7 0x00000aab4cb4 in alternator::expressionsParser::path() build/debug/gen/alternator/expressionsParser.cpp:694:29
    #8 0x00000aac0ca4 in alternator::expressionsParser::projection_expression() build/debug/gen/alternator/expressionsParser.cpp:1983:15
    #9 0x00000340a8a4 in do_with_parser(std::basic_string_view<char, std::char_traits<char>>) test/boost/alternator_prepared_expression_cache_test.cc:197:26
    #10 0x00000340a8a4 in memory_leak::run_test_case() const test/boost/alternator_prepared_expression_cache_test.cc:215:9
    #11 0x7ff455070050 in seastar::testing::seastar_test::run()::$_0::operator()() const build/debug/seastar/./seastar/src/testing/seastar_test.cc:51:16
    #12 0x7ff45506ff2f in seastar::future<void> std::__invoke_impl<seastar::future<void>, seastar::testing::seastar_test::run()::$_0&>(std::__invoke_other, seastar::testing::seastar_test::run()::$_0&) /usr/lib/gcc/x86_64-redhat-linux/15/../../../../include/c++/15/bits/invoke.h:63:14
    #13 0x7ff45506fe3f in std::enable_if<is_invocable_r_v<seastar::future<void>, seastar::testing::seastar_test::run()::$_0&>, seastar::future<void>>::type std::__invoke_r<seastar::future<void>, seastar::testing::seastar_test::run()::$_0&>(seastar::testing::seastar_test::run()::$_0&) /usr/lib/gcc/x86_64-redhat-linux/15/../../../../include/c++/15/bits/invoke.h:116:9
    #14 0x7ff45506faf7 in std::_Function_handler<seastar::future<void> (), seastar::testing::seastar_test::run()::$_0>::_M_invoke(std::_Any_data const&) /usr/lib/gcc/x86_64-redhat-linux/15/../../../../include/c++/15/bits/std_function.h:292:9
    #15 0x7ff45986f590 in std::function<seastar::future<void> ()>::operator()() const /usr/lib/gcc/x86_64-redhat-linux/15/../../../../include/c++/15/bits/std_function.h:593:9
    #16 0x7ff45508c404 in seastar::testing::test_runner::run_sync(std::function<seastar::future<void> ()>)::$_0::operator()() const build/debug/seastar/./seastar/src/testing/test_runner.cc:139:20
    #17 0x7ff45508c20f in seastar::future<void> std::__invoke_impl<seastar::future<void>, seastar::testing::test_runner::run_sync(std::function<seastar::future<void> ()>)::$_0&>(std::__invoke_other, seastar::testing::test_runner::run_sync(std::function<seastar::future<void> ()>)::$_0&) /usr/lib/gcc/x86_64-redhat-linux/15/../../../../include/c++/15/bits/invoke.h:63:14
    #18 0x7ff45508c0ff in std::enable_if<is_invocable_r_v<seastar::future<void>, seastar::testing::test_runner::run_sync(std::function<seastar::future<void> ()>)::$_0&>, seastar::future<void>>::type std::__invoke_r<seastar::future<void>, seastar::testing::test_runner::run_sync(std::function<seastar::future<void> ()>)::$_0&>(seastar::testing::test_runner::run_sync(std::function<seastar::future<void> ()>)::$_0&) /usr/lib/gcc/x86_64-redhat-linux/15/../../../../include/c++/15/bits/invoke.h:116:9
    #19 0x7ff45508bb37 in std::_Function_handler<seastar::future<void> (), seastar::testing::test_runner::run_sync(std::function<seastar::future<void> ()>)::$_0>::_M_invoke(std::_Any_data const&) /usr/lib/gcc/x86_64-redhat-linux/15/../../../../include/c++/15/bits/std_function.h:292:9
    #20 0x7ff45986f590 in std::function<seastar::future<void> ()>::operator()() const /usr/lib/gcc/x86_64-redhat-linux/15/../../../../include/c++/15/bits/std_function.h:593:9
    #21 0x7ff455088a65 in seastar::testing::test_runner::start_thread(int, char**)::$_0::operator()()::'lambda'()::operator()() const::'lambda0'()::operator()() const::'lambda0'()::operator()() const build/debug/seastar/./seastar/src/testing/test_runner.cc:94:28
    #22 0x7ff45508886f in seastar::future<void> seastar::futurize<seastar::future<void>>::invoke<seastar::testing::test_runner::start_thread(int, char**)::$_0::operator()()::'lambda'()::operator()() const::'lambda0'()::operator()() const::'lambda0'()&>(seastar::testing::test_runner::start_thread(int, char**)::$_0::operator()()::'lambda'()::operator()() const::'lambda0'()::operator()() const::'lambda0'()&) build/debug/seastar/./seastar/include/seastar/core/future.hh:1959:20
    #23 0x7ff45508815f in auto seastar::futurize_invoke<seastar::testing::test_runner::start_thread(int, char**)::$_0::operator()()::'lambda'()::operator()() const::'lambda0'()::operator()() const::'lambda0'()&>(seastar::testing::test_runner::start_thread(int, char**)::$_0::operator()()::'lambda'()::operator()() const::'lambda0'()::operator()() const::'lambda0'()&) build/debug/seastar/./seastar/include/seastar/core/future.hh:1990:12
    #24 0x7ff455087aa2 in seastar::future<void> seastar::do_until<seastar::testing::test_runner::start_thread(int, char**)::$_0::operator()()::'lambda'()::operator()() const::'lambda0'()::operator()() const::'lambda0'(), seastar::testing::test_runner::start_thread(int, char**)::$_0::operator()()::'lambda'()::operator()() const::'lambda0'()::operator()() const::'lambda'()>(T0, T) build/debug/seastar/./seastar/include/seastar/core/loop.hh:350:18
    #25 0x7ff45508780e in seastar::testing::test_runner::start_thread(int, char**)::$_0::operator()()::'lambda'()::operator()() const::'lambda0'()::operator()() const build/debug/seastar/./seastar/src/testing/test_runner.cc:90:22
    #26 0x7ff45508765f in seastar::future<void> seastar::futurize<seastar::future<void>>::invoke<seastar::testing::test_runner::start_thread(int, char**)::$_0::operator()()::'lambda'()::operator()() const::'lambda0'()&>(seastar::testing::test_runner::start_thread(int, char**)::$_0::operator()()::'lambda'()::operator()() const::'lambda0'()&) build/debug/seastar/./seastar/include/seastar/core/future.hh:1959:20
    #27 0x7ff4550875ef in auto seastar::futurize_invoke<seastar::testing::test_runner::start_thread(int, char**)::$_0::operator()()::'lambda'()::operator()() const::'lambda0'()&>(seastar::testing::test_runner::start_thread(int, char**)::$_0::operator()()::'lambda'()::operator()() const::'lambda0'()&) build/debug/seastar/./seastar/include/seastar/core/future.hh:1990:12
    #28 0x7ff455087583 in auto seastar::future<void> seastar::future<void>::then<seastar::testing::test_runner::start_thread(int, char**)::$_0::operator()()::'lambda'()::operator()() const::'lambda0'(), seastar::future<void>>(T&&)::'lambda'(auto&&...)::operator()<>(auto&&...) build/debug/seastar/./seastar/include/seastar/core/future.hh:1368:24
    #29 0x7ff4550874e7 in seastar::noncopyable_function<seastar::future<void> ()>::direct_vtable_for<seastar::future<void> seastar::future<void>::then<seastar::testing::test_runner::start_thread(int, char**)::$_0::operator()()::'lambda'()::operator()() const::'lambda0'(), seastar::future<void>>(T&&)::'lambda'(auto&&...)>::call(seastar::noncopyable_function<seastar::future<void> ()> const*) build/debug/seastar/./seastar/include/seastar/util/noncopyable_function.hh:138:20
    #30 0x7ff4596ca2a6 in seastar::noncopyable_function<seastar::future<void> ()>::operator()() const build/debug/seastar/./seastar/include/seastar/util/noncopyable_function.hh:224:16
    #31 0x7ff459704f5f in seastar::future<void> std::__invoke_impl<seastar::future<void>, seastar::noncopyable_function<seastar::future<void> ()>&>(std::__invoke_other, seastar::noncopyable_function<seastar::future<void> ()>&) /usr/lib/gcc/x86_64-redhat-linux/15/../../../../include/c++/15/bits/invoke.h:63:14
    #32 0x7ff459704eef in std::__invoke_result<seastar::noncopyable_function<seastar::future<void> ()>&>::type std::__invoke<seastar::noncopyable_function<seastar::future<void> ()>&>(seastar::noncopyable_function<seastar::future<void> ()>&) /usr/lib/gcc/x86_64-redhat-linux/15/../../../../include/c++/15/bits/invoke.h:98:14
    #33 0x7ff459704e7f in std::invoke_result<seastar::noncopyable_function<seastar::future<void> ()>&>::type std::invoke<seastar::noncopyable_function<seastar::future<void> ()>&>(seastar::noncopyable_function<seastar::future<void> ()>&) /usr/lib/gcc/x86_64-redhat-linux/15/../../../../include/c++/15/functional:122:14
    #34 0x7ff459704ce3 in auto seastar::internal::future_invoke<seastar::noncopyable_function<seastar::future<void> ()>&, seastar::internal::monostate>(seastar::noncopyable_function<seastar::future<void> ()>&, seastar::internal::monostate&&) build/debug/seastar/./seastar/include/seastar/core/future.hh:1131:16

SUMMARY: AddressSanitizer: 88 byte(s) leaked in 1 allocation(s).
```

Strangely for all printable ASCII only `.,[` cause memory leak for an expression `fmt::format("a[0{}b]", x)`.
The rule that is parsed there, is that inside `[ ]` there must be one or more digits.
Before I dig deeper into it, @nyh do you have any quick guesses?

### PR DESCRIPTION
**Pull request by m-szymon**
This patch overrides the antlr3 function that allocates the missing tokens that would eventually leak. The override stores these tokens in a vector, ensuring memory is freed whenever the parser is destroyed. Solution is copied from CQL implementation.

Fixes #25878

Fixes memory leak - should be backported

### CODE REVIEW

**Comment by nyh** [comment]
Thanks for looking into this issue. Good catch that CQL had the same problem, and was fixed in 26a3461908dcef3ee622124e726d2691dcad3767.

So if I understand correctly, every time that ANTLR3 wants to report a syntax error it allocates these "missing tokens" that it wants to report, and it **never** ever freed them, even when no exceptions were involved? How could they miss this bug? I wonder how did **we** miss this bug until now in our tests - we do have tests for Alternator expression errors, for example in test/alternator/test_update_expression.py. Is it possible that in debug build we enable leak detection only in C++ unit tests, but not in test/alternator tests?

Also, I wonder if any of the ideas I raised in the issue regarding throwing can also cause bugs. Now that these "missing tokens" are safe from leakage, can something else leak because of the use of expressions? I suggest we add to the test the expression `a[9999999999999999999]` and see what happens on the integer overflow.

File: alternator/expressions.g [hunk 1]
```diff
@@ -91,6 +91,16 @@ options {
         throw expressions_syntax_error(format("{} at char {}", err,
             ex->get_charPositionInLine()));
     }
+
+    // ANTLR3 tries to recover missing tokens, but it has a bug and leaks them.
```
Comment (nyh) [inline alternator/expressions.g]: What do you mean "recover" missing tokens? I would use the verb "report", not "recover".

File: configure.py [hunk 2]
```diff
@@ -447,6 +447,7 @@ def find_ninja():
     'test/boost/UUID_test',
     'test/boost/advanced_rpc_compressor_test',
     'test/boost/allocation_strategy_test',
+    'test/boost/alternator_expressions_test',
```
Comment (nyh) [inline configure.py]: I think you should consider putting the new tests inside the existing alternator_unit_test instead of creating a separate source file. alternator_unit_test is one of the smallest unit test source files we have, and the new test file is also tiny - there's no risk it's becoming too big. In the past we used to create a separate executable for each test which made it very expensive to have many small test files. I don't know if that's true any more, but in any case I don't see a benefit of splitting a file that developers will normally want to run in its entirety anyway.

File: test/boost/alternator_expressions_test.cc [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+/*
+ * SPDX-License-Identifier: LicenseRef-ScyllaDB-Source-Available-1.0
+ */
+
+#include <boost/test/unit_test.hpp>
+#include <seastar/testing/test_case.hh>
+#include <seastar/core/coroutine.hh>
+#include "alternator/expressions.hh"
+
+// ANTLR3 leaks memory when it tries to recover from missing token.
```
Comment (nyh) [inline test/boost/alternator_expressions_test.cc]: Again, I think the relevant verb is "report".

**Comment by m-szymon** [comment]
@scylladbbot trigger-ci

File: test/boost/alternator_expressions_test.cc [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    BOOST_REQUIRE_THROW(alternator::parse_update_expression("ADD a v"), alternator::expressions_syntax_error); // missing ':'
+    
+    BOOST_REQUIRE_THROW(alternator::parse_condition_expression("size(a < 5", "Test"), alternator::expressions_syntax_error); // missing ')'
+    BOOST_REQUIRE_THROW(alternator::parse_condition_expression("a IN :x)", "Test"), alternator::expressions_syntax_error); // missing '('
+    BOOST_REQUIRE_THROW(alternator::parse_condition_expression("a IN (:x", "Test"), alternator::expressions_syntax_error); // missing ')'
+    BOOST_REQUIRE_THROW(alternator::parse_condition_expression("a BETWEEN :x AN :y", "Test"), alternator::expressions_syntax_error); // missing 'AND'
+    BOOST_REQUIRE_THROW(alternator::parse_condition_expression("a BETWEEN :x :y", "Test"), alternator::expressions_syntax_error); // missing 'AND'
+
+    BOOST_REQUIRE_THROW(alternator::parse_projection_expression("a[0.b"), alternator::expressions_syntax_error); // missing ']'
+    co_return;
```
Comment (nyh) [inline test/boost/alternator_expressions_test.cc]: A thought: Can you please try to add another test function (it's not a "missing token" issue) for parsing `a[99999999999999999]`? I want to see that despite the throw in stoi, we still report the error correctly and don't leak. Or - if it does leak, we'll have a reproducer for a new issue ;-)

File: test/boost/alternator_expressions_test.cc [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+ * SPDX-License-Identifier: LicenseRef-ScyllaDB-Source-Available-1.0
+ */
+
+#include <boost/test/unit_test.hpp>
+#include <seastar/testing/test_case.hh>
+#include <seastar/core/coroutine.hh>
+#include "alternator/expressions.hh"
+
+// ANTLR3 leaks memory when it tries to recover from missing token.
+// Reproduces #25878
```
Comment (nyh) [inline test/boost/alternator_expressions_test.cc]: Thanks for adding a comment to explain the test.
I would add to it a sentence that this leak was reported by ASAN (?) when running this test in debug mode - the test passed but the leak is discovered when the test file exits (right?).

**Comment by nyh** [comment]
> @scylladbbot trigger-ci

Before doing that, you are supposed to look at the failures and "prove" that they are flaky tests and not your "fault". You usually do this by either finding an existing open issue complaining about this test as being flaky (and also adding a mention of the new occurrence in the existing test), or if there is no such issue and you truely believe this is a flaky test and not the fault of your patch - you should open a new issue. I'll do this now for you:

> Failed Tests (1/74826):
>    [test_view_building_with_tablet_move.debug.1](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/19707/testReport/junit/cluster.mv/test_mv_building/debug___test_py___debug___test_view_building_with_tablet_move_debug_1) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_view_building_with_tablet_move.debug.1)

This is a known flaky test, see https://github.com/scylladb/scylladb/issues/25909.

File: alternator/expressions.g (hunk 1 above)
Comment (m-szymon) [reply alternator/expressions.g]: ANTLR calls it "recover" - it is not just for reporting. It tries to finish parsing and create valid objects, as if the missing token was there.
I can write exactly that.
I didn't think more details are necessary here, and "recover missing tokens" gives a good context if anyone wants to find it in ANTLR.

File: configure.py (hunk 2 above)
Comment (m-szymon) [reply configure.py]: I had in mind that I have PR with other expression related tests. As I expect this one to be merged quite quickly, I planned to extend this file in next PR.
But it is still relatively small number of test, so do you suggest that in general I should stick to alternator_unit_test?

File: test/boost/alternator_expressions_test.cc (hunk 5 above)
Comment (m-szymon) [reply test/boost/alternator_expressions_test.cc]: yes, done.

File: test/boost/alternator_expressions_test.cc (hunk 4 above)
Comment (m-szymon) [reply test/boost/alternator_expressions_test.cc]: I checked and it reports std::out_of_range and doesn't leak.
But I will add separate issue for that case, as I think it is still not correct.
I believe it should be ignored, as comment in python tests says:
```
    # If some of the paths are not available, they are silently ignored (just
    # like they returned an empty item when used alone earlier)
```
I think it should be covered in python tests, as we have such tests there.
Unless python test can't detect memory leaks, but in that case we may want to add more cases.

File: test/boost/alternator_expressions_test.cc (hunk 3 above)
Comment (m-szymon) [reply test/boost/alternator_expressions_test.cc]: It is recovery.

**Review by m-szymon — COMMENTED** [review]
> So if I understand correctly, every time that ANTLR3 wants to report a syntax error it allocates these "missing tokens" that it wants to report, and it never ever freed them, even when no exceptions were involved?

No. Only when there is no alternative token possible and it looks ahead if the following token could be acceptable, assuming we missed that one. It creates "fake" token and reports the situation, but without our exception it would continue parsing.

>  How could they miss this bug? 

Shame.

>I wonder how did we miss this bug until now in our tests - we do have tests for Alternator expression errors, for example in test/alternator/test_update_expression.py. 

It is not that often case. The bigger grammar and more alternatives possible - the less chance of seeing it. I couldn't find more examples for alternator.
Still probably we don't test that much syntax errors. Most cases are either valid or fail at resolving.

> Is it possible that in debug build we enable leak detection only in C++ unit tests, but not in test/alternator tests?

I don't know that - I tried to see leaks from python, but without success.

> > @ scylladbbot trigger-ci
>
> Before doing that, you are supposed to look at the failures ...

Thanks. I know, I quickly checked and was 99% sure it is not my fault. I just wanted to restarted CI, before I start working. If I noticed you already started review, I would have wait with it.

File: test/boost/alternator_expressions_test.cc (hunk 4 above)
Comment (m-szymon) [reply test/boost/alternator_expressions_test.cc]: Well, DynamoDB also has some limit - it just gives better message.
Nevertheless it is a candidate for python test.

I will add also here, with other cases that trigger exception.

File: alternator/expressions.g (hunk 1 above)
Comment (nyh) [reply alternator/expressions.g]: Ok, I wasn't familiar with this ANTLR "recovery" terminology. Interesting. Maybe it also explains why even in the syntax error `a[0.b]` we got an stoi() call - the "." is an unexpected token and maybe ANTLR tried to "recover" by adding a "]" and then continuing to run stoi(0).

File: configure.py (hunk 2 above)
Comment (nyh) [reply configure.py]: Yes, in my opinion yes - I think if we have in alternator_unit_test roughly 100 lines of code, and you plan to add roughly 100 more lines of code, it can be the same test file.
Having the same test file makes it easier to run all the Alternator tests, faster and smaller to compile (especially if separate test executables are used, I'm not sure this is the case), and generally easier to just read the entire file and see what is being tested.

File: test/boost/alternator_expressions_test.cc (hunk 4 above)
Comment (nyh) [reply test/boost/alternator_expressions_test.cc]: Very interesting. I agree with your view that if `a[999]` is supposed to return nothing because the list doesn't have 999 items, then certainly `a[999999999999999]` can just return nothing - and not fail.

However, it's still customary for parsers to parse first and only then check for other errors, so it's not surprising for the parsing of 999999999999999 to fail early before realizing it could have just done nothing. Let's try to do the same thing that DynamoDB does in this error, unless we have a reason not to. You're right that this needs a separate test in Python, and is not really related to this patch. I can do that.

Comment (nyh) [reply test/boost/alternator_expressions_test.cc]: https://github.com/scylladb/scylladb/issues/25947

Comment (nyh) [reply test/boost/alternator_expressions_test.cc]: I'm still really confused how exceptions from the parser work and don't leak a lot of stuff, not just that "missing token", but apparently, it works? Please add the `a[999999999999999999]` to your C++ leak test as well, to see it really doesn't leak. Thanks.

**Comment by m-szymon** [comment]
> #### Failed Tests (1/74838):
> 
>     * [test_view_building_scheduling_group.release.1](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/19712/testReport/junit/cluster.mv/test_mv_building/release___test_py___release___test_view_building_scheduling_group_release_1) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_view_building_scheduling_group.release.1)

#25957

**Comment by m-szymon** [comment]
@nyh can you merge it? I want it my next PR.

**Comment by m-szymon** [comment]
@scylladb/scylla-maint please merge.

**Comment by nyh** [comment]
> @scylladb/scylla-maint please merge.

Done
